Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -893,35 +860,59 @@ private class ProjectStructure
             public readonly ProjectInfo ProjectInfo;
             private readonly Context context;
 
-            public ProjectStructure(Context context, string projectSpecificConfigDir, string projectSpecificOutputDir, ProjectConfig projectConfig, ProjectInfo projectInfo)
+            public ProjectStructure(Context context, string projectSpecificConfigDir, string projectSpecificOutputDir, bool checkAndLoadConfigFile)
             {
                 this.context = context;
                 ProjectSpecificConfigDir = projectSpecificConfigDir;
                 ProjectSpecificOutputDir = projectSpecificOutputDir;
-                ProjectConfig = projectConfig;
-                ProjectInfo = projectInfo;
+
+                var projectInfoFile = TryAddToResults(projectSpecificOutputDir, FileConstants.ProjectInfoFileName); // should always exist
```

Comment:
nitpick: this line is a bit too long to have an inline comment

I think the comment can be removed, as `AssertFileExists(projectInfoFile);` is just below?

it would make sense to explain "why" it is always there in a comment above